macOS/DuckDuckGoVPN/VPNAppEventsHandler.swift

```diff
@@ -36,10 +41,19 @@ final class VPNAppEventsHandler {
             versionStore.lastAgentVersionRun = currentVersion
         }
 
-        let restartTunnel = {
+        let restartTunnel = { [appState, tunnelController] in
             self.tunnelController.ensureRiskyDomainsEnabledIfNeeded()
             Logger.networking.info("App updated from \(versionStore.lastAgentVersionRun ?? "null", privacy: .public) to \(currentVersion, privacy: .public): updating login items")
-            self.restartTunnel()
+
+            if tunnelController.settings.isAuthV2Enabled && !appState.isMigratedToAuthV2 {
```

Comment:
To make testing easier I recommend adding `appState.isMigratedToAuthV2 = false` in the `init` for this class.